### PR TITLE
#2135-execution-result-naming - use file name and create generic dataset

### LINKED ISSUE #2135: [VIP] Better manage execution result naming
Currently, result files are named after execution's name, which is not quite clear.
Use file name directly.

### PR DESCRIPTION
**Pull request by jcomedouteau**
Closes #2135 

How to test:
- Make an execution on VIP
--> Result processed dataset are named after file name
--> Created processed dataset is generic and not mesh (???)

### CODE REVIEW

**Comment by youennmerel** [comment]
Hi @jcomedouteau,

As discussed we sould name the processing as the corresponding pipeline.

> :gear: OFSEP_sequence_identifier/0.2
>    |-- :camera: output.json

Cheers,

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java [hunk 1]
```diff
@@ -229,16 +229,30 @@ private void reconvertToNifti(String format, HttpServletResponse response, Long
 		List<String> files = new ArrayList<>();
 		for (File res : workFolder.listFiles()) {
 
+			String datasetFilePath = res.getAbsolutePath();
+			String fileName = res.getName();
+			boolean compressed = datasetFilePath.endsWith(".nii");
```
Comment (youennmerel) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java:234]: Maybe this variable should be call "toCompress" (or "toZip") ?

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java [hunk 2]
```diff
@@ -229,16 +229,30 @@ private void reconvertToNifti(String format, HttpServletResponse response, Long
...
+
+			// Gzip file if necessary in order to always return a .nii.gz file
+			if (compressed) {
+				datasetFilePath = datasetFilePath + ".gz";
```
Comment (youennmerel) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java:238]: Maybe use a constant for the ".gz" zip extension

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/controller/ExecutionApiController.java [hunk 3]
```diff
@@ -202,6 +202,7 @@ private String getInputValueUri(ExecutionCandidateDTO candidate, String groupBy,
                 + "?format=" + exportFormat
                 + "&resourceId=" + resourceId
                 + "&token=" + authenticationToken
+                + (candidate.getConverterId()  != null? ("&converter=" + candidate.getConverterId()) : "")
```
Comment (youennmerel) [inline shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/controller/ExecutionApiController.java:205]: Add a space before `?` here

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java (hunk 2 above)
Comment (jcomedouteau) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java]: Done!

File: shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java (hunk 1 above)
Comment (jcomedouteau) [reply shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java]: Done

**Review by youennmerel — APPROVED** [review]

**Pull request merged**